pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/client/Auth0ManagementClient.java

```diff
@@ -538,8 +540,20 @@ private <B, E> ApiResult<B, E> withRetries(String retryMessage, Supplier<ApiResu
                 break;
             }
             if (res.getStatusCode() == 429) {
-                log.error(retryMessage, res.getError());
+                log.error(retryMessage, res.getError() + " " + res.getError().getClass().getName());
                 long wait = backoffMillis * numTries + new Random().nextInt(MAX_JITTER_MILLIS);
+                // if we have more information from auth0 about how long to wait, use it.
+                if (res.getError() instanceof RateLimitException) {
+                    RateLimitException rateLimit = (RateLimitException) res.getError();
+                    long unixTimeAtWhichToRetry = rateLimit.getReset();
+                    long suggestedWaitTime = unixTimeAtWhichToRetry - Instant.now().getEpochSecond();
+                    if (suggestedWaitTime > 0) {
+                        wait = suggestedWaitTime * 1000;
+                    }
+                    log.warn("Hit auth0 rate limit.  Pausing for " + wait + "s based on auth0 headers.");
+                } else {
+                    log.warn("Hit auth0 rate limit.  Pausing for " + wait + "ms");
+                }
```

Comment:
In the warning message below, perhaps add the amount of time you waited. Just a debugging aid.

Reply:
Logging wait time total.

Reply:
Looks like seconds in the first message?

Reply:
Renamed few variables hopefully clarify things a little.
Variables `unixTimeAtWhichToRetry` and `suggestedWaitTime` are time in seconds. Rest are time in milliseconds.